### PR TITLE
Covidcast upload early exit

### PR DESCRIPTION
**Pull request by krivard**
**WARNING** this PR includes a commit reducing the size of the covid_hosp_meta.revision_timestamp column from 1024 to 512. Nothing depends on that change except compliance with MariaDB maximum key length, but once this is merged we should make that change in prod. No value in that column is presently more than 23 characters, so shouldn't be an issue.

**Actual purpose:** Covidcast Acquisition currently counts the number of rows in the `covidcast` table regardless of whether the `receiving` directory is empty. This operation now takes upwards of 40 minutes, which is silly.

This PR rearranges the upload script somewhat so that it first checks `receiving`, then exits as a no-op if it finds no files.

In the process, I merged the default and with-specific-issue routines.

Unit tests updated and pass.

~Integration tests I seem to have forgot about. I'll update here when they're ready.~

Integration tests work automagically because they use no mocks.

Added a test to verify early-exit.

### CODE REVIEW

**Comment by krivard** [comment]
Tagging George to check my merge of the with/without specific issue stuff

File: src/acquisition/covidcast/csv_to_database.py [hunk 1]
```diff
@@ -167,33 +136,38 @@ def archive_as_successful(path_src, filename, source):
...
-    scan_upload_archive_impl=scan_upload_archive,
-    scan_issue_specific_impl=scan_issue_specific):
+    collect_files_impl=collect_files,
+    upload_archive_impl=upload_archive):
   """Find, parse, and upload covidcast signals."""
 
   if args.is_wip_override and args.not_wip_override:
     print('conflicting overrides for forcing WIP option!  exiting...')
-    exit()
+    return
```
Comment (melange396) [inline src/acquisition/covidcast/csv_to_database.py:148]: good call (pun intended)

File: src/acquisition/covidcast/csv_to_database.py [hunk 2]
```diff
@@ -32,107 +32,70 @@ def get_argument_parser():
...
-  data_dir: top-level directory where CSVs are stored
-  database: an open connection to the epidata database
-
-  The CSV storage layout is assumed to be as follows:
+  :path_details: output from CsvImporter.find*_csv_files 
+  
+  :database: an open connection to the epidata database
 
-  - Receiving: <data_dir>/receiving/<source name>/*.csv
-  - Archival: <data_dir>/archive/<status>/<source name>/*.csv[.gz]
```
Comment (melange396) [inline src/acquisition/covidcast/csv_to_database.py:105]: do we care about preserving this note about the directory structure?

**Review by melange396 — APPROVED** [review]
LGTM

File: src/acquisition/covidcast/csv_to_database.py (hunk 2 above)
Comment (krivard) [reply src/acquisition/covidcast/csv_to_database.py:105]: I thought about that, but it's in this file as well: https://github.com/cmu-delphi/delphi-epidata/blob/main/src/acquisition/covidcast/data_dir_readme.md#layout

& that gets dropped into /common/receiving

**Pull request merged**